### PR TITLE
Fix unable to pass extra parameters to psycopg2 in Postgres connector

### LINKED ISSUE #5442: [BUG] Unable to pass extra parameters to psycopg2 in Postgres connector
### Mage version

0.9.71

### Describe the bug

Trying to set different parameters for the psycopg2 connection does not work, because the Postgres class only sends a limited set of parameters when initiating a connection

From https://github.com/mage-ai/mage-ai/blob/master/mage_ai/io/postgres.py#L131
```
            connect_opts = dict(
                database=database,
                host=host,
                password=password,
                port=port,
                user=user,
                keepalives=1,
                keepalives_idle=300,
            )
  
            self._ctx = connect(**connect_opts)
```

For example, changing the `sslmode` parameter for psycopg2 is impossible with the current implementation

### To reproduce

Example usage to show the issue with the parameter

```
   with Postgres(dbname = credentials['dbname'],
                    user = credentials['user'],
                    password = credentials['password'],
                    host = credentials['host'],
                    port = credentials['port'],
                    sslmode = 'disable') as loader:
        print(f"\nloader: {loader.conn. get_dsn_parameters()}")

```
Output:
```
loader: {'user': 'someuser', 'channel_binding': 'prefer', 'dbname': '*************', 'host': '***********************************************************************************************', 'port': '1111', 'options': '', 'keepalives': '1', 'keepalives_idle': '300', 'sslmode': 'prefer', 'sslcompression': '0', 'sslsni': '1', 'ssl_min_protocol_version': 'TLSv1.2', 'gssencmode': 'prefer', 'krbsrvname': 'postgres', 'target_session_attrs': 'any'}

```
The `sslmode` is `prefer` instead of `disable`.

### Expected behavior

Expected output:
```
loader: {'user': 'someuser', 'channel_binding': 'prefer', 'dbname': '*************', 'host': '***********************************************************************************************', 'port': '1111', 'options': '', 'keepalives': '1', 'keepalives_idle': '300', 'sslmode': 'disable', 'sslcompression': '0', 'sslsni': '1', 'ssl_min_protocol_version': 'TLSv1.2', 'gssencmode': 'prefer', 'krbsrvname': 'postgres', 'target_session_attrs': 'any'}
```

### Screenshots

_No response_

### Operating system

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kanenorman**
# Description

This fix allows  all valid psycopg2 connection parameters as keyword arguments. Previously, only a limited set of parameters were passed to connect(), restricting customization options like `sslmode` and `keepalives`. The fix now allows users to specify any connection parameter, which is then properly passed to `connect()`, enhancing flexibility in PostgreSQL connections.

fixes #5442

# How Has This Been Tested?

Manually tested and confirmed output. Did not see existing test for the class.
Open to writing test if requested.

```python
from mage_ai.io.postgres import Postgres

credentials = {
    'dbname': 'postgres',
    'user': 'postgres',
    'password': 'postgres',
    'host': 'localhost',
    'port': '5432',
    'sslmode': 'disable'
}

with Postgres(dbname = credentials['dbname'],
                user = credentials['user'],
                password = credentials['password'],
                host = credentials['host'],
                port = credentials['port'],
                sslmode = 'disable',
                keepalives=15,
                keepalives_idle=500,
                connect_timeout=10,
                verbose=True,
                ) as loader:
    print(f"\nloader: {loader.conn.get_dsn_parameters()}")

# Postgres initialized
#└─ Opening connection to PostgreSQL database...DONE
# loader: {'user': 'postgres', 'channel_binding': 'prefer', 'connect_timeout': '10', 'dbname': 'postgres', 'host': 'localhost', 'port': '5432', 'options': '', 'keepalives': '15', 'keepalives_idle': '500', 'sslmode': 'disable', 'sslcompression': '0', 'sslsni': '1', 'ssl_min_protocol_version': 'TLSv1.2', 'gssencmode': 'prefer', 'krbsrvname': 'postgres', 'target_session_attrs': 'any'}

```
**Mage Version**: 0.9.74
**Python**: 3.11.3
**OS**: Mac Sequoia 15

# Checklist
- [ ] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

cc:

@wangxiaoyou1993

### CODE REVIEW

**Comment by Kuki145** [comment]
Just checking, will this be merged anytime soon?

**Comment by kanenorman** [comment]
> Just checking, will this be merged anytime soon?

@wangxiaoyou1993

**Comment by wangxiaoyou1993** [comment]
> > Just checking, will this be merged anytime soon?
> 
> @wangxiaoyou1993

Sorry, forgot to review. Let me check.

**Comment by Kuki145** [comment]
@wangxiaoyou1993 Any updates?

**Comment by kanenorman** [comment]
> @wangxiaoyou1993 Any updates?

@wangxiaoyou1993

File: mage_ai/io/postgres.py [hunk 1]
```diff
@@ -128,19 +130,35 @@ def open(self) -> None:
...
+                'schema',
+                'connection_method',
+                'ssh_host',
+                'ssh_port',
+                'ssh_username',
+                'ssh_password',
+                'ssh_pkey',
+                "verbose"
+            ]
+            for key in unrecognized_keys:
```
Comment (wangxiaoyou1993) [inline mage_ai/io/postgres.py:147]: can you make sure the key exists before pops it?

**Review by wangxiaoyou1993 — APPROVED** [review]

**Pull request merged**